pkg/controller/cloud/node_controller.go

```diff
@@ -191,25 +191,22 @@ func (cnc *CloudNodeController) updateNodeAddress(node *v1.Node, instances cloud
 	}
 }
 
+// UpdateCloudNode handles updating nodes registered with the cloud taint.
 func (cnc *CloudNodeController) UpdateCloudNode(_, newObj interface{}) {
-	node, ok := newObj.(*v1.Node)
-	if !ok {
-		utilruntime.HandleError(fmt.Errorf("unexpected object type: %v", newObj))
-		return
-	}
-
-	cloudTaint := getCloudTaint(node.Spec.Taints)
-	if cloudTaint == nil {
```

Comment:
The only difference is the log line
```
klog.V(2).Infof("This node %s is registered without the cloud taint. Will not process.", node.Name)
```

Which I think is okay to duplicate for that reason because Update events are much more frequent we removed it for Update but we still want it for Add. What do you think?

Reply:
What about modifying `handleCloudNode` to
```golang
func (cnc *CloudNodeController) handleCloudNode(obj interface{}, update bool) {
...
```
if `update` is false, it can print some extra log. 🤔

Reply:
I'm not sure that abstraction is worth it just for 1 log line, I think the small duplication is simpler, what do you think?

Reply:
If this kind of duplication is acceptable, I will close this issue in few days~